mcstatus/__main__.py

```diff
@@ -52,35 +78,38 @@ def status(server: SupportedServers) -> int:
 
 
 def json(server: SupportedServers) -> int:
-    data = {}
-    data["online"] = False
-    data["kind"] = server.kind()
-    # Build data with responses and quit on exception
+    data = {"online": False, "kind": server.kind()}
+
+    status_res = query_res = None
     try:
         status_res = server.status(tries=1)
-        java_res = status_res if isinstance(status_res, JavaStatusResponse) else None
-        data["version"] = status_res.version.name
-        data["protocol"] = status_res.version.protocol
-        data["motd"] = status_res.motd.to_minecraft()
-        data["player_count"] = status_res.players.online
-        data["player_max"] = status_res.players.max
-        data["players"] = []
-        if java_res and java_res.players.sample is not None:
-            data["players"] = [{"name": player.name, "id": player.id} for player in java_res.players.sample]
-
-        data["ping"] = status_res.latency
-        data["online"] = True
-
         if isinstance(server, JavaServer):
             query_res = server.query(tries=1)
-            data["host_ip"] = query_res.raw["hostip"]
-            data["host_port"] = query_res.raw["hostport"]
-            data["map"] = query_res.map
-            data["plugins"] = query_res.software.plugins
-    except Exception:  # TODO: Check what this actually excepts
-        pass
-
-    print(json_dumps(data))
+    except Exception as e:
+        if status_res is None:
+            data["error"] = str(e)
+
+    # construct 'data' dict outside try/except to ensure data processing errors
+    # are noticed.
+    data["online"] = bool(status_res or query_res)
+    if status_res is not None:
+        data["status"] = dataclasses.asdict(status_res)
+
+        # XXX: hack to fixup MOTD serialisation. should be implemented elsewhere.
```

Comment:
What do you mean implemented elsewhere?

Reply:
There should be a method which serialises MOTD objects to a sensible JSON representation, and it should not be within the CLI.

Reply:
I don't really understand how you imagine this

```py
def to_json(self) -> dict:
    return {"raw": self.to_minecraft()}
```

Why? If you absolutely need MOTD in JSON format, you can use `.raw`, but because of so many implementations, it is hard to parse so it is easier to parse minecraft (string) representation.

Reply:
I guess the question is whether the mcstatus means to / wants to have an official JSON format, in which case it would make sense to have methods in the class instead of at every use. If you'd like, I can remove this comment.

Reply:
I don't see a reason to have JSON API inside our Python API, if a user wants to export answer of a server as JSON, they can just build a [wrapper](https://github.com/StatusMC/backend/tree/ce3ed289f72fcac5281537c2606c4375c0ccdf8e/src/models/v1) or use `dataclasses.asdict` without any problem.

On the other side, having JSON API in CLI is completely reasonable.

Reply:
My point is that they _can't_ just use dataclasses.asdict() with no problems, because of this required fixup. In any case, I've removed the comment.

Reply:
That's a good idea, I brought it up on discord

https://discord.com/channels/936788458939224094/938623566964981802/1264998810850168963